caravel/assets/javascripts/components/ModalTrigger.jsx

```diff
@@ -0,0 +1,55 @@
+import React, { PropTypes } from 'react';
+import { Modal } from 'react-bootstrap';
+
+const propTypes = {
+  buttonBody: PropTypes.node.isRequired,
+  modalTitle: PropTypes.string.isRequired,
+  modalBody: PropTypes.node.isRequired,
+  beforeOpen: PropTypes.func,
+};
+
+const defaultProps = {
+  beforeOpen: () => {},
+};
+
+export default class ModalTrigger extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showModal: false,
+    };
+    this.open = this.open.bind(this);
+    this.close = this.close.bind(this);
+  }
+
+  close() {
+    this.setState({ showModal: false });
+  }
+
+  open() {
+    this.props.beforeOpen();
+    this.setState({ showModal: true });
+  }
+
+  render() {
+    return (
+      <span
```

Comment:
agreed, let's call it `triggerNode` and take out the `btn` styling so it can be used more generally.